ai-assistant/src/ai/prompts.ts

```diff
@@ -12,9 +12,25 @@ MANDATORY TOOL USAGE:
 - If a user asks for current data from the cluster (pods, deployments, services, etc.), you MUST call kubernetes_api_request
 - Example: "show me pods" → MUST call kubernetes_api_request(url="/api/v1/pods", method="GET")
 - Example: "list pods in default namespace" → MUST call kubernetes_api_request(url="/api/v1/namespaces/default/pods", method="GET")
-- Example: "get logs for pod-name" → MUST call kubernetes_api_request(url="/api/v1/namespaces/default/pods/pod-name/log", method="GET")
+- Example: "get logs for pod-name" → First get pod details to check containers, then fetch logs with container parameter if needed
 - Do NOT just say "I'll fetch the data" - actually call the tool immediately
 
+LOG HANDLING FOR MULTI-CONTAINER PODS:
+- When a user asks for logs from a pod, ALWAYS first check the pod specification to determine the number of containers
+- If the pod has only one container, directly fetch logs: kubernetes_api_request(url="/api/v1/namespaces/default/pods/pod-name/log", method="GET")
+- If the pod has multiple containers, you MUST ask the user which container they want logs from
+- List the available container names and ask the user to specify
+- Once the user specifies a container, fetch logs with the container parameter: kubernetes_api_request(url="/api/v1/namespaces/default/pods/pod-name/log?container=container-name", method="GET")
+- NEVER attempt to fetch logs from a multi-container pod without specifying the container name
+- Examples of user requests that specify containers:
+  - "get logs from pod-name container nginx" → kubernetes_api_request(url="/api/v1/namespaces/default/pods/pod-name/log?container=nginx", method="GET")
+  - "show logs for container sidecar in pod-name" → kubernetes_api_request(url="/api/v1/namespaces/default/pods/pod-name/log?container=sidecar", method="GET")
+- If you encounter an error about "container name must be specified", the error handler will automatically provide guidance to the user
+- IMPORTANT: Parse user requests carefully to detect if they're specifying a container name in their request:
+  - Look for patterns like "container [name]", "from container [name]", "[pod-name] [container-name]"
+  - If user specifies a container name, use it directly in the log URL
+  - If user doesn't specify a container and the pod has multiple containers, fetch pod details first to list available containers
```

Comment:
Again, seems like you are assuming the kubernetes tool is enabled and this seems like something that should mostly belong there?

Reply:
fixed